### PR TITLE
Storages: Only print errno instead of its string when CPU affinity of SegmentReader bind fail

### PR DESCRIPTION
**Pull request by JinheLin**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tiflash/issues/6233

Problem Summary: printing the strerrno may cause data race under the new CI pipeline
```
      0x1141614e	StackTrace::StackTrace() [gtests_dbms+289497422]
       0x7e6baa4	fault_signal_handler(int) [gtests_dbms+132561572]
  0x7f054334c400	<unknown symbol> [libc.so.6+222208]
  0x7f054334f2bd	getenv [libc.so.6+234173]
  0x7f0543346e2e	__dcigettext [libc.so.6+200238]
  0x7f05433a2cde	strerror_r [libc.so.6+576734]
       0xdf52693	DB::DM::SegmentReader::setCPUAffinity() [gtests_dbms+234170003]
       0xdf52329	DB::DM::SegmentReader::run() [gtests_dbms+234169129]
       0xdf55de9	decltype(*(static_cast<DB::DM::SegmentReader*>(fp0)).*fp()) std::__1::__invoke<void (DB::DM::SegmentReader::*)(), DB::DM::SegmentReader*, void>(void (DB::DM::SegmentReader::*&&)(), DB::DM::SegmentReader*&&) [gtests_dbms+234184169]
       0xdf55d6e	void std::__1::__thread_execute<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, void (DB::DM::SegmentReader::*)(), DB::DM::SegmentReader*, 2ul>(std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, void (DB::DM::SegmentReader::*)(), DB::DM::SegmentReader*>&, std::__1::__tuple_indices<2ul>) [gtests_dbms+234184046]
       0xdf55a82	void* std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, void (DB::DM::SegmentReader::*)(), DB::DM::SegmentReader*> >(void*) [gtests_dbms+234183298]
  0x7f0543901ea5	start_thread [libpthread.so.0+32421]
  0x7f0543414b0d	clone [libc.so.6+1043213]
```

### What is changed and how it works?

Only print the errno instead of its string

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by purelind** [comment]
> /debug pull-unit-test

The pipeline triggered in the current debug mode will not report the task link. The current triggered pipeline link is:   https://do.pingcap.net/jenkins/blue/organizations/jenkins/pingcap%2Ftiflash%2Fpull_unit_test/detail/pull_unit_test/11/pipeline

File: dbms/src/Storages/DeltaMerge/ReadThread/SegmentReader.cpp [hunk 1]
```diff
@@ -141,7 +115,9 @@ class SegmentReader
 
     void run()
     {
-        setCPUAffinity();
+#ifndef DBMS_PUBLIC_GTEST // Does not bind CPUs in gtest.
+        setCPUAffinity(cpus, log);
+#endif
```
Comment (JaySon-Huang) [inline dbms/src/Storages/DeltaMerge/ReadThread/SegmentReader.cpp:120]:
```suggestion
        setCPUAffinity(cpus, log);
```

File: dbms/src/Storages/DeltaMerge/tests/gtest_dm_delta_merge_store.cpp [hunk 2]
```diff
@@ -1561,7 +1561,9 @@ CATCH
 TEST_P(DeltaMergeStoreRWTest, IngestDupHandleVersion)
 try
 {
-    setStorageFormat(5); // Some old formats does not support ingest DMFiles.
+    // Some old formats does not support ingest DMFiles.
+    if (mode == TestMode::V1_BlockOnly)
+        return;
```
Comment (JaySon-Huang) [inline dbms/src/Storages/DeltaMerge/tests/gtest_dm_delta_merge_store.cpp:1566]: comes from https://github.com/pingcap/tiflash/pull/8737#discussion_r1469656420

**Review by JaySon-Huang — APPROVED** [review]

**Review by Lloyd-Pottiger — APPROVED** [review]

**Comment by Lloyd-Pottiger** [comment]
/cherry-pick-release-7.5

**Pull request merged**